cmk/base/plugins/agent_based/cisco_asa_sensors.py

```diff
@@ -0,0 +1,363 @@
+    }
+    return states.get(st, State.CRIT)
+
+
+def parse_cisco_asa_sensors(string_table: List[StringTable]) -> CiscoAsaSensors:
+    temp = {}
+    fan = {}
+    power = {}
+
+    for sensorname, sensortype, sensorvalue, sensorstatus, sensorunits in string_table[0]:
+        if sensorname != '':  # for asa context, there are no real sensors.
+            if sensortype == '8':  # Temperature
+                temp.update({sensorname.replace('Temperature ', ''): CiscoAsaTempSensor(
+                    value=to_celsius(float(sensorvalue), sensorunits),
+                    unit=sensorunits,
+                    state=get_sensor_state(sensorstatus),
+                    status=int(sensorstatus) - 1,
+                    status_readable=get_status_readable(sensorstatus),
+                )})
+
+            if sensortype == '10':  # Fan
+                fan.update({sensorname.replace('Fan ', ''): CiscoAsaFanSensor(
+                    value=int(sensorvalue),
+                    unit=sensorunits,
+                    state=get_sensor_state(sensorstatus),
+                    status_readable=get_status_readable(sensorstatus),
+                )})
+
+            if sensortype == '12':  # Power supply
+                power.update({sensorname.replace('Power ', ''): CiscoAsaPowerSensor(
+                    state=get_sensor_state(sensorstatus),
+                    status_readable=get_status_readable(sensorstatus),
+                )})
+
+    return CiscoAsaSensors(temp=temp, fan=fan, power=power)
+
+
+register.snmp_section(
+    name='cisco_asa_sensors',
+    parse_function=parse_cisco_asa_sensors,
+    fetch=[
+        SNMPTree(
+            base='.1.3.6.1.2.1',  #
+            oids=[
+                '47.1.1.1.1.7',  # ENTITY-MIB::entPhysicalName
+                '99.1.1.1.1',  # ENTITY-SENSOR-MIB::entPhySensorType
+                '99.1.1.1.4',  # ENTITY-SENSOR-MIB::entPhySensorValue
+                '99.1.1.1.5',  # ENTITY-SENSOR-MIB::entPhySensorOperStatus
+                '99.1.1.1.6',  # ENTITY-SENSOR-MIB::entPhySensorUnitsDisplay
+            ]
+        ),
+    ],
+    detect=startswith('.1.3.6.1.2.1.1.1.0', 'cisco adaptive security appliance')
+)
+
+
+# ##################################################################################################
+#
+# ASA SENSORS TEMPERATURE
+#
+# ##################################################################################################
+
+
+def discovery_cisco_asa_temp(section: CiscoAsaSensors) -> DiscoveryResult:
+    yield from (Service(item=item) for item in section.temp.keys())
+
+
+def check_cisco_asa_temp(item, params: TempParamType, section: CiscoAsaSensors) -> CheckResult:
+    try:
+        sensor = section.temp[item]
+    except KeyError:
+        return
+
+    yield Result(state=sensor.state, summary='Status: %s' % sensor.status_readable)
+
+    yield from check_temperature(
+        sensor.value,
+        dev_unit=sensor.unit,
+        dev_status=sensor.status,
```

Comment:
I started to merge this PR. I would suggest that we remove the dev_status and dev_status_name arguments here, since we already yield a result with the sensor status above, therefore, this is redundant. Would that be ok for you? I can do this on the way while merging.

Reply:
This is fine with me. The only thing, the can also the status field from CiscoAsaTempSensor be removed, an so on.. If you like i can do this...

Reply:
Thats ok, since its only a small change, I will take care of that on the way